performances/performance_test_plugin_cmake/cmake/generate_script.py

```diff
@@ -141,6 +141,46 @@ def get_sub_factory(msgs, package):
     subscribers_factory.at(msg_type)();
   }
 
+  extern "C" void add_subscriber_impl_lifecycle(
+    std::shared_ptr<performance_test::LifecycleNode> n,
```

Comment:
All the code that is in this file is quite a terrible hack.
I don't see the point of duplicating all of it.

We should rather just have 2 different strings for this second line as it looks the only difference between the old and the new version.
So you can call every python function twice passing this node type as argument

Reply:
I'm not sure what you mean by "2 different strings"? Here we have to pass in the pointer to the node

Reply:
This code does not know anything about pointers or nodes, it only works with strings.

Functions like `get_sub_factory()` are generating the code for a C++ function as a string.
You duplicated the content of the whole function just to generate a different string where the only difference is that line, i.e. in one case you have 
```
std::shared_ptr<performance_test::Node> n,
```
and in the other
```
std::shared_ptr<performance_test::LifecycleNode> n,
```

if the type of node was a parameter, you could just call the function twice.
```
content += get_sub_factory(msgs, package, "LifecycleNode")
content += get_sub_factory(msgs, package, "Node")
```

Reply:
I've modified the python script to take a string for the node type as an argument, however, because of limitations of C code, I can't add function overloads, so we still will have separate API created (though now it is less duplicated code).